src/include/duckdb/execution/operator/aggregate/physical_ungrouped_aggregate.hpp

```diff
@@ -22,6 +45,8 @@ class PhysicalUngroupedAggregate : public PhysicalOperator {
 
 	//! The aggregates that have to be computed
 	vector<unique_ptr<Expression>> aggregates;
+	//! The data used for the distinct aggregates (if any)
+	DistinctAggregateData distinct_aggregate_data;
```

Comment:
This should not be part of the actual operator - but should be stored in the states - at least the radix hash tables. Anything that can be changed should not be stored in the operator as the operator can be re-used multiple times (in the case of recursive CTEs or prepared statements).

Reply:
Even the table? I don't see how I can get the states from the table in the Local state this way, if I store the table in the Global state, it's not passed as a parameter to GetLocalSinkState

Reply:
`op.sink_state` should be present when `GetLocalSinkState` is called, no?